.github/workflows/test.yml

```diff
@@ -16,13 +16,37 @@ jobs:
     - run: npm test
 
   # test action works running from the graph
-  test:
+  test_bcc:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        runningBranch: [bcc, kind, microshift]
     steps:
     - uses: actions/checkout@v3
     - uses: ./
       with:
-        runningBranch: ${{ matrix.runningBranch }}
+        runningBranch: bcc
+    - name: verify
+      run: |
+        dpkg -l | grep bcc
+
+  # test action works running from the graph
+  test_kind:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ./
+      with:
+        runningBranch: kind
+    - name: verify
+      run: |
+        dpkg -l | grep bcc
+        docker ps -a
+        kind get kubeconfig --name=kind
+        kubectl get pods
+
+  # test action works running from the graph
+  test_microshift:
+    runs-on: ubuntu-latest
```

Comment:
microshift isn't supported on ubuntu - only rhel. I will look into [self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) to see if this is possible. `test_microshift` might be a follow-up but will be super nice to have (and I'm sure microshift project will also be interested in this)

Reply:
en.... 
if microshift want to make a github action for specific self-hosted runners, I am not sure that part of code should be in kepler or not?
for ex a microshift based github action with Tekon?

otherwise switch/communication between self-host runners and github pre defined runners seems a magic thing.
as build docker image on pre defined ubuntu runners and push image to self-host runners.

I am not sure if pre defined host has enough resources as Tekon + Promethes/OTeL operator + kepler ... running all of them in kind?

Reply:
or a short term solution is that we running all microshift related test at self-hosted rhel runner.